lib/Doctrine/Common/Inflector/Inflector.php

```diff
@@ -402,10 +404,13 @@ public static function pluralize($word)
             self::$plural['cacheIrregular']   = '(?:' . implode('|', array_keys(self::$plural['merged']['irregular'])) . ')';
         }
 
-        if (preg_match('/(.*)\\b(' . self::$plural['cacheIrregular'] . ')$/i', $word, $regs)) {
-            self::$cache['pluralize'][$word] = $regs[1] . substr($word, 0, 1) . substr(self::$plural['merged']['irregular'][strtolower($regs[2])], 1);
-            
-            return self::$cache['pluralize'][$word];
+        if (preg_match('/(.*(?:\\b|_))(' . self::$plural['cacheIrregular'] . ')$/i', $tablized_word, $regs)) {
+            $result = $regs[1] . substr($regs[2], 0, 1) . substr(self::$plural['merged']['irregular'][strtolower($regs[2])], 1);
+            if ($tablized_word !== $word) {
```

Comment:
Naming is a bit weird - is this the ubiquitous language here? I really have no idea myself :-\